msticpy/context/azure/sentinel_ti.py

```diff
@@ -0,0 +1,455 @@
+            A list of URLs that provide an external reference for the indicator
+        valid_from : datetime, optional
+            A datetime from which the indicator is valid from, defaults to now
+        valid_to : datetime, optional
+            A datetime to which the indicator is valid until
+
+        Raises
+        ------
+        MsticpyUserError
+            If invalid ioc_type or confidence value provided
+        CloudError
+            If API call fails
+
+        """
+        self.check_connected()  # type: ignore
+        ti_url = self.sent_urls["ti"] + "/createIndicator"  # type: ignore
+        params = {"api-version": "2021-10-01"}
+        if ioc_type not in [
+            "domain-name",
+            "url",
+            "SHA-256",
+            "SHA-1",
+            "MD5",
+            "ipv4-addr",
+            "ipv6_addr",
+        ]:
+            raise MsticpyUserError(
+                """ioc_type must be one of -
+                 'domain-name', 'url',  'SHA-256', 'SHA-1', 'MD5', 'ipv4-addr', 'ipv6_addr'"""
+            )
+        pattern_type = ioc_type
+        value = "value"
+        if ioc_type in ["SHA-256", "SHA-1", "MD5"]:
+            pattern_type = "file"
+            value = f"hashes.'{ioc_type}'"
+
+        if confidence > 100 or confidence < 0:
+            raise MsticpyUserError("confidence must be between 0 and 100")
+        data_items = {
+            "displayName": name,
+            "confidence": confidence,
+            "pattern": f"[{pattern_type}:{value} = '{indicator}']",
+            "patternType": pattern_type,
+            "revoked": "false",
+            "source": "MSTICPy",
+        }
+        additional_data_items = _build_additional_indicator_items(**kwargs)
+        data_items.update(additional_data_items)
+        data = _build_sent_data(data_items, props=True)
+        data["kind"] = "indicator"
+        response = httpx.post(
+            ti_url,
+            headers=get_api_headers(self.token),  # type: ignore
+            params=params,
+            content=str(data),
+            timeout=get_http_timeout(),
+        )
+        if response.status_code not in (200, 201):
+            raise CloudError(response=response)
+        if not silent:
+            print("Indicator created.")
+
+        return response.json().get("name")
+
+    def bulk_create_indicators(
+        self,
+        data: pd.DataFrame,
+        indicator_column: str = "Observable",
+        indicator_type_column: str = "IoCType",
+        **kwargs,
+    ):
+        """
+        Bulk create indicators from a DataFrame.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            A dataframe containing indicators and indicator types
+        indicator_column : str, optional
+            The column containing indicator values to create, by default "Observable"
+        indicator_type_column : str, optional
+            The column containing indicator type values, by default "IoCType"
```

Comment:
Might want to add the acceptable types here.
Mapping from Msticpy types here might be messier. I guess we could create a function that you could use in an apply or maybe assign.
```python
_SENTINEL_IOC_TYPES = {
    "ipv4": "ipaddr_v4",
    # etc.
}

def map_to_sentinel_ioc_type(ioc_type: str):
    if ioc_type in _SENTINEL_IOC_TYPES.values():
        return ioc_type
    return _SENTINEL_IOC_TYPES.get(ioc_type)
```
or something like that